test/api.js

```diff
@@ -244,6 +250,38 @@ describe('/v1', function() {
       });
     });
 
+    describe('untrusted client scope', function() {
+      it('should fail if invalid scope', function() {
+        var client = clientByName('Untrusted');
+        mockAssertion().reply(200, VERIFY_GOOD);
+        return Server.api.post({
+          url: '/authorization',
+          payload: authParams({
+            client_id: client.id,
+            scope: 'profile:write'
+          })
+        }).then(function(res) {
+          assert.equal(res.statusCode, 400);
+          assert.equal(res.result.errno, 114);
+          assert.equal(res.result.invalidScopes[0], 'profile:write');
+        });
+      });
+
+      it('should succeed if valid scope', function() {
+        var client = clientByName('Untrusted');
+        mockAssertion().reply(200, VERIFY_GOOD);
+        return Server.api.post({
+          url: '/authorization',
+          payload: authParams({
+            client_id: client.id,
+            scope: 'profile:email profile:uid'
+          })
+        }).then(function(res) {
+          assert.equal(res.statusCode, 200);
+        });
+      });
+    });
+
```

Comment:
It's probably worth adding a test for a multi-scope request like "profile profile:write" or similar, just to be doubly sure.